api/v1alpha3/azuremachine_types.go

```diff
@@ -0,0 +1,136 @@
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cluster-api/errors"
+)
+
+const (
+	// MachineFinalizer allows ReconcileAzureMachine to clean up Azure resources associated with AzureMachine before
+	// removing it from the apiserver.
+	MachineFinalizer = "azuremachine.infrastructure.cluster.x-k8s.io"
+)
+
+// AzureMachineSpec defines the desired state of AzureMachine
+type AzureMachineSpec struct {
+	// ProviderID is the unique identifier as specified by the cloud provider.
+	// +optional
+	ProviderID *string `json:"providerID,omitempty"`
+
+	VMSize           string           `json:"vmSize"`
+	AvailabilityZone AvailabilityZone `json:"availabilityZone,omitempty"`
+
+	Image *Image `json:"image,omitempty"`
+
+	OSDisk OSDisk `json:"osDisk"`
+
+	Location string `json:"location"`
+
+	SSHPublicKey string `json:"sshPublicKey"`
+
+	// AdditionalTags is an optional set of tags to add to an instance, in addition to the ones added by default by the
+	// Azure provider. If both the AzureCluster and the AzureMachine specify the same tag name with different values, the
+	// AzureMachine's value takes precedence.
+	// +optional
+	AdditionalTags Tags `json:"additionalTags,omitempty"`
+
+	// AllocatePublicIP allows the ability to create dynamic public ips for machines where this value is true.
+	// +optional
+	AllocatePublicIP bool `json:"allocatePublicIP,omitempty"`
+}
+
+// AzureMachineStatus defines the observed state of AzureMachine
+type AzureMachineStatus struct {
+	// Ready is true when the provider resource is ready.
+	// +optional
+	Ready bool `json:"ready"`
+
+	// Addresses contains the Azure instance associated addresses.
+	Addresses []v1.NodeAddress `json:"addresses,omitempty"`
+
+	// VMState is the provisioning state of the Azure virtual machine.
+	// +optional
+	VMState *VMState `json:"vmState,omitempty"`
+
+	// ErrorReason will be set in the event that there is a terminal problem
+	// reconciling the Machine and will contain a succinct value suitable
+	// for machine interpretation.
+	//
+	// This field should not be set for transitive errors that a controller
+	// faces that are expected to be fixed automatically over
+	// time (like service outages), but instead indicate that something is
+	// fundamentally wrong with the Machine's spec or the configuration of
+	// the controller, and that manual intervention is required. Examples
+	// of terminal errors would be invalid combinations of settings in the
+	// spec, values that are unsupported by the controller, or the
+	// responsible controller itself being critically misconfigured.
+	//
+	// Any transient errors that occur during the reconciliation of Machines
+	// can be added as events to the Machine object and/or logged in the
+	// controller's output.
+	// +optional
+	ErrorReason *errors.MachineStatusError `json:"errorReason,omitempty"`
```

Comment:
`ErrorReason` was renamed to `FailureReason` in [v1alpha3](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md#external-objects-will-need-to-rename-statuserrorreason-and-statuserrormessage).

```suggestion
	FailureReason *errors.MachineStatusError `json:"failureReason,omitempty"`
```

Reply:
Thanks for reviewing @noamran, will make the changes